src/types/qwseat.cpp

```diff
@@ -371,7 +396,11 @@ bool QWSeat::touchHasGrab() const
 
 void QWSeat::touchNotifyCancel(QWSurface *surface)
 {
+#if WLR_VERSION_MINOR > 17
+    wlr_seat_touch_notify_cancel(handle(), wlr_seat_client_from_resource(surface->handle()->resource));
```

Comment:
add a new function,  we shoudn't just call 
`wlr_seat_client_from_resource` here

Reply:
that's following the suggestion in wlroots
> Note, `wl_resource_get_client()` and `wlr_seat_client_for_wl_client()` can be used to get a `struct wlr_seat_client` from a `struct wlr_surface`.